src/main/resources/pip-inspector.py

```diff
@@ -128,43 +127,66 @@ def populate_dependency_tree(project_root_node, requirements_path):
 
 def recursively_resolve_dependencies(package_name, history):
     """Forms a DependencyNode by recursively resolving its dependencies. Tracks history for cyclic dependencies."""
-    package = get_package_by_name(package_name)
+    dependency_node, child_names = get_package_by_name(package_name)
 
-    if package is None:
+    if dependency_node is None:
         return None
 
-    dependency_node = DependencyNode(package.project_name, package.version)
-
-    if package_name.lower() not in history:
-        history.append(package_name.lower())
-        for package_dependency in package.requires():
-            child_node = recursively_resolve_dependencies(package_dependency.key, history)
+    if dependency_node.name not in history:
+        history.append(dependency_node.name)
+        for child_name in child_names:
+            child_node = recursively_resolve_dependencies(child_name, history)
             if child_node is not None:
                 dependency_node.children = dependency_node.children + [child_node]
 
     return dependency_node
 
+try: # attempt to import and rely on importlib.metadata which has been available since Python 3.8
+    import importlib.metadata
 
-def get_package_by_name(package_name):
-    """Looks up a package from the pip cache"""
-    if package_name is None:
-        return None
-
-    package_dict = working_set.by_key
-    try:
-        # TODO: By using pkg_resources.Requirement.parse to get the correct key, we may not need to attempt the other
-        #     methods. Robust tests are needed to confirm.
-        return package_dict[Requirement.parse(package_name).key]
-    except:
-        pass
-
-    name_variants = (package_name, package_name.lower(), package_name.replace('-', '_'), package_name.replace('_', '-'))
-    for name_variant in name_variants:
-        if name_variant in package_dict:
-            return package_dict[name_variant]
-
-    return None
+    def get_package_by_name(package_name):
+        if package_name is None:
+            return None, None
 
+        try:
+            metadata = importlib.metadata.metadata(package_name.strip())
+        except importlib.metadata.PackageNotFoundError:
+            return None, None
+
+        dependency_node = DependencyNode(metadata["Name"], metadata["Version"])
+
+        requirement_names = []
+        requirements = importlib.metadata.requires(dependency_node.name)
```

Comment:
I discovered a problem with this approach of listing dependencies since the `requires()` function can return packages which are not necessarily installed: in cases when the dependency is an "extra" or condition such as `platform_python_implementation != 'PyPy'` is evaluated to `False`.

I am currently exploring possible ways of filtering out such dependencies.